### PR TITLE
automate screenshots

### PR DESCRIPTION
**Pull request by brindy**
Reviewer: Mia, Jeffrey
Asana: https://app.asana.com/0/414235014887631/390649588301838/f
and fixes a bug for: https://app.asana.com/0/414235014887631/413609835171504/f
CC:

**Description**:

Automates screenshot creation for a range of iPhones and iPads

**Steps to test this PR**:

1. With fastlane installed run `fastlane screenshots`
2. Once complete a browser will open with a html file showing each of the screenshots

3. Launch the app normally and explicitly enter a search url which overrides the location parameter, e.g. https://duckduckgo.com/q=privacy&kl=ie-en should show results for Ireland 
4. After the page loads the URL should contain the mobile stats params

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR DRI Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications (Database connections, Grafana stats, CPU)

### CODE REVIEW

File: Core/AppUrls.swift [hunk 1]
```diff
@@ -95,15 +95,18 @@ public struct AppUrls {
...
      */
     public func searchUrl(text: String) -> URL {
-        let searchUrl = home
-            .addParam(name: Param.search, value: text)
-            .addParam(name: Param.source, value: ParamValue.source)
+        let searchUrl = home.addParam(name: Param.search, value: text)
+        return fixStatsParams(for: searchUrl)
+    }
+
+    public func fixStatsParams(for url: URL) -> URL {
```
[the remark this answers is not in the document]
Comment (brindy) [reply Core/AppUrls.swift]: Will do

File: DuckDuckGo.xcodeproj/xcshareddata/xcschemes/ContentBlockerExtension.xcscheme [hunk 2]
```diff
@@ -43,13 +43,23 @@
...
+         </TestableReference>
       </Testables>
       <MacroExpansion>
          <BuildableReference
             BuildableIdentifier = "primary"
-            BlueprintIdentifier = "F128109E1EB35B5300F95E4E"
-            BuildableName = "ContentBlockerExtension.appex"
-            BlueprintName = "ContentBlockerExtension"
+            BlueprintIdentifier = "84E341911E2F7EFB00BDBA6F"
+            BuildableName = "DuckDuckGo.app"
```
[the remark this answers is not in the document]
Comment (brindy) [reply DuckDuckGo.xcodeproj/xcshareddata/xcschemes/ContentBlockerExtension.xcscheme]: It's just the ordering of the targets in the project, but will double check.

File: DuckDuckGo.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -592,6 +612,7 @@
 				F15A8B751E43A428008B8C6F /* ShareExtension */,
 				F12810A01EB35B5300F95E4E /* ContentBlockerExtension */,
 				84E341A91E2F7EFB00BDBA6F /* UnitTests */,
+				857AEF801F4F2AE9003B84A1 /* UITests */,
```
[the remark this answers is not in the document]
Comment (brindy) [reply DuckDuckGo.xcodeproj/project.pbxproj]: Agreed

File: DuckDuckGo/AppDelegate.swift [hunk 4]
```diff
@@ -47,7 +51,11 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         startMigration()
         StatisticsLoader.shared.load()
         TrackerLoader.shared.updateTrackers()
-        startOnboardingFlowIfNotSeenBefore()
+
+        if !isUITest {
+            startOnboardingFlowIfNotSeenBefore()
```
[the remark this answers is not in the document]
Comment (brindy) [reply DuckDuckGo/AppDelegate.swift]: Maybe. It's tricky to test if the onboading is showing and then decide to dismiss it as we would only see it the first time the app is launched. I might be able to check for the presence of the onboarding screen's text as part of the setup and then dismiss it. 

Launch arguments are not an usual thing (the same approach is often used to supress normal app launch for unit tests), but I agree it feels hacky.

**Comment by brindy** [comment]
re: Setting the time, I've already created an Asana task for that:

https://app.asana.com/0/414709148257752/415242964614490/f

File: DuckDuckGo/AppDelegate.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (brindy) [reply DuckDuckGo/AppDelegate.swift]: OK, done. :)

File: DuckDuckGo.xcodeproj/xcshareddata/xcschemes/ContentBlockerExtension.xcscheme (hunk 2 above)
[the remark this answers is not in the document]
Comment (brindy) [reply DuckDuckGo.xcodeproj/xcshareddata/xcschemes/ContentBlockerExtension.xcscheme]: When I added the UITest target Xcode (helpfully) added the testable reference to these schemes. Removed now.

File: DuckDuckGo.xcodeproj/project.pbxproj (hunk 3 above)
[the remark this answers is not in the document]
Comment (brindy) [reply DuckDuckGo.xcodeproj/project.pbxproj]: Done

File: Core/AppUrls.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (brindy) [reply Core/AppUrls.swift]: Done

File: AppScreenshots/AppScreenshotsUITests.swift [hunk 5]
```diff
@@ -0,0 +1,163 @@
+//
+//  UITests.swift
```
Comment (subsymbolic) [inline AppScreenshots/AppScreenshotsUITests.swift]: Another one. Sorry!

File: fastlane/Fastfile [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+# we automatically generate. Feel free to remove this line
+# once you get things running smoothly!
+generated_fastfile_id "c7ddc28b-25d3-424d-b688-e693165a4521"
+
+default_platform :ios
+
+lane :screenshots do
+
+  gym(
+    scheme: "UITests",
```
Comment (subsymbolic) [inline fastlane/Fastfile]: The old scheme name is still here and `fastlane screenshots` isn't running

File: fastlane/Snapfile [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  "iPhone 7",
+  "iPad Pro (12.9 inch)",
+  "iPad Pro (9.7 inch)"
+])
+
+languages([
+  "en-US",
+])
+
+scheme "UITests"
```
Comment (subsymbolic) [inline fastlane/Snapfile]: Old scheme reference

**Comment by brindy** [comment]
Sorry - ran them as UI tests and didn't run the fastlane lane.  🤦‍♂️ 

Made the changes and ran it to confirm it's all correct.

**Pull request merged**